analysis/test/association_test.R

```diff
@@ -57,14 +57,26 @@ SamplePopulations <- function(N, num_variables = 1, params,
                       function(i) rep(1:m, ceiling(N / m))[1:N])
   } else {
     # Otherwise, draw from a Poisson random variable
-    variables <- lapply(1:num_variables, function(i) rpois(N, 1))
+    variables <- lapply(1:num_variables, function(i) rpois(N, 1)+1)
     if (!variable_opts$independent) {
-      # If user wants dependent RVs, take the cumulative sum of 3 independent
-      #     random variables.
-      variables <- as.list(data.frame(t(apply(do.call("rbind",
-                                                      variables),
-                                              2, cumsum))))
-      # Use the same cohort assignment in all 3 dimensions so the
+      # If user wants dependent RVs, take the cumulative sum of num_variables
+      # independent random variables.
+      # variables <- as.list(data.frame(t(apply(do.call("rbind",
+      #                                                variables),
+      #                                        2, cumsum))))
+
+      # Otherwise, draw the first variable from a Poisson distribution (+1
+      # to avoid 0 index errors)
+      # Subsequent variables are closely correlated with the first variable
+      # in the foll. manner
+      #   variable_i ~ variable_1 + (i-1).U(0, 1)
```

Comment:
Better write as (i - 1) * Bernoulli(.5) (it's better to avoid using '.' for multiplication since R treats dots as part of tokens).